### DISCUSSION TITLE
SDPA format in pyhon

### DISCUSSION THREAD

**Discussion by yangbo116** [sdpa-python/sdpa-python#2] [General]
I use the SDPA python (https://sdpa-python.github.io/docs/usage/) version to generate a SDPA format as the Sparse Input Data File in user Manual, but I get the follow format:
3
1
2S
-4.8000000000000000e+01 +8.0000000000000000e+00 -2.0000000000000000e+01
0 1 1 1 +1.1000000000000000e+01
0 1 2 2 -2.3000000000000000e+01
1 1 1 1 -1.0000000000000000e+01
1 1 1 2 -4.0000000000000000e+00
2 1 2 2 -4.0000000000000000e+00
3 1 1 2 +8.0000000000000000e+00
3 1 2 2 +2.0000000000000000e+00
, which is not consistent with the results in user Manual:
"Example 1: mDim = 3, nBLOCK = 1, {2}"
3 = mDIM
1 = nBLOCK
2 = bLOCKsTRUCT
{48, -8, 20}
0 1 1 1 -11
0 1 2 2 23
1 1 1 1 10
1 1 1 2 4
2 1 2 2 -8
3 1 1 2 -8
3 1 2 2 -2
, at the m=2, 1 block, （2, 2): value -8， but in python，this position is -4. So, is anything wrong in python code?

**Comment by usamamuneeb** [comment]
**Jan 2025 edit**: Since the sign conventions of the `.dat-s` read/write routines were not intuitive to the user, as of version 0.2.2, the `fromsdpa` routine is being deprecated in favor of `importsdpa`. It's accompanied by `exportsdpa` (which replaces the existing `tosdpa` routine in the original SDPAP code).

Upon both import and export, the new routines will automatically perform sign flipping (which is a consequence of the inverse relationship between SDPA and SeDuMi formats).
